Improve the regex of sed command of SVN_SERF_INCLUDES macro

The regex of sed command is too simple, which mangles any "-D"
in strings, for example, a project name is "My-Demo", it will
be removed from the path by the sed regex, which causes a build
failure. Improving it to only delete the -D behind of a space or 
the -D at start of a string.

Reference

ttps://bugzilla.yoctoproject.org/show_bug.cgi?id=7874

Upstream-Status: Pending

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
Signed-off-by: Yue Tao <yue.tao@windriver.com>

diff --git a/build/ac-macros/serf.m4.old b/build/ac-macros/serf.m4
index 885c5a1..39969c6 100644
--- a/build/ac-macros/serf.m4.old
+++ b/build/ac-macros/serf.m4
@@ -143,7 +143,7 @@ AC_DEFUN(SVN_SERF_PKG_CONFIG,
         if $PKG_CONFIG $serf_major --atleast-version=$serf_check_version; then
           AC_MSG_RESULT([yes])
           serf_found=yes
-          SVN_SERF_INCLUDES=[`$PKG_CONFIG $serf_major --cflags | $SED -e 's/-D[^ ]*//g'`]
+          SVN_SERF_INCLUDES=[`$PKG_CONFIG $serf_major --cflags | $SED -e 's/[[:space:]]-D[^ ]*//g' -e 's/^-D[^ ]*//g'`]
           SVN_SERF_LIBS=`$PKG_CONFIG $serf_major --libs` 
           break
         else
